rules.js

```diff
@@ -186,6 +186,103 @@ class sub_with_return extends rule(3) {
     }
 }
 
+class object_formatting extends rule(4) {
+    constructor() {
+        super()
+        this._node = 'object'
+        this.message = ``
+        this.spaces_after_colon = 1
+        this.spaces_before_colon = 0
+        this.trailing_commas = 'no' // 'no', 'required'
+        this.maxOneLine = 4
+        this.key_in_quotes = 'consistent' // 'allways', 'consistent', 'any'
+    }
+
+    check(node) {
+        let warnings = []
+        let quotedKeys = 0
+
+        const newlineCount = (token) => {
+            if (token.type != 'NL') return 0
```

Comment:
why this type has capital `'NL'` where others have lowercase `ws` ?

Reply:
Artifact of lexer/parser token naming I think. lexer tokens traditionally have all caps names. Although I would have to look into why whitespace token is in lowercase